src/main/java/seedu/programmer/ui/MainWindow.java

```diff
@@ -243,8 +243,8 @@ public void handleShowResult() {
      */
     @FXML
     private void handleDashboard() {
+        dashboardWindow.update();
```

Comment:
Would it be better to move the update so that is called after every command? 

Currently the update is done in `executeCommand()` only if the dashboard is showing. We could remove the `isShowing()` condition and remove this `update()` call here and I think the dashboard should still update dynamically.

Similar to how the app data is saved after each command?

Reply:
Think this is a similar situation to how the app data is saved after each command

Reply:
Sorry for the late reply, I think the issue here is that handleDashboard can be called by pressing the Dashboard button, which bypasses the execute command part.

I agree with your suggestion that we need to remove the `isShowing` condition for executeCommand and I will change it in another commit.

Reply:
Right, I didn't account for the button. Ok, sounds good. Let me know when you have pushed the commit 👍